_posts/2019-12-25-technical-debt.md

```diff
@@ -0,0 +1,51 @@
+---
+layout: post
+title:  "Technical Debt"
+date:   2019-08-05
+tags: debt devops dora 
+author: Matt Tyler
+image:
+---
+
+> Debt . . . . that peculiar nexus where money, narrative or story, and religious belief intersect, often with explosive force. - Margaret Atwood
+
+Technical debt is an infamous subject in software development. The common understanding is that we accrue it by some series of faustian pacts; short-cuts are taken in order to deliver a feature sooner, with the underlying assumption that additional features will take longer to complete. A development team that continues to accrue technical debt will eventually find time taken to deliver the next feature will begin to approach infinity. In defiance of what is an obvious problem to the average developer, business stakeholders are often loathe to address it. Technical debt is unyielding to measure and containment due to it being a vague concept. If developers wish redress from technical debt, a better way must be found to disclose and appraise the value.
+
+There is a long list of potential consequences for allowing the unfettered accrual of technical debt. A reduction in the ability to deliver features (usually referred to as 'velocity') is chief among them. Knowing the consequences of even the smallest revision to the system is difficult. Minor changes accord ample opportunity to observe the butterfly effect. At the core of this is understanding. A lack of understanding has consequences beyond knowing how to safely modify the system. It handicaps our ability to prevent outages. When those outages occur (and they will), our ignorance will frustrate any attempts at triage.
+
+> i have a theory, which is that we struggle to get the time allocated to pay down technical debt (or improve deploys, etc) because to biz types we basically sound like the underpants gnomes. step 1 ... pay down technical debt. step 2 ... step 3 .. PROFIT - Charity Majors, CTO @ Honeycomb.io
```

Comment:
I agree with Charity, but I think it's even more simple. If we described it in $'s, it would be compelling. A failure to do so means that 'business types' will never appreciate the cost.